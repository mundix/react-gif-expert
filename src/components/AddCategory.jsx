import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

   const [inputValue, setInputValue] = useState('')

   const onInputChange = ({ target }) => {
      // console.log(event.target.value)
      // setInputValue(event.target.value);
      setInputValue(target.value);
   }

   const onSubmit = (event) => {
      // no quiero que se haga refreshc uando se hace enter
      event.preventDefault();
      // console.log(inputValue);
      // Cuando usas el setCategory del useState, el mantiene los valores de por si 
      // es comoel ejemplo del setCounter( (c) => c + 1); hace referencia a sus valores si lo volvemos como argumento 
      // de funcion de flecha 
      
      if(inputValue.trim().length <= 1) return;

      // setCategories((categories) => [inputValue, ...categories]); //esto es si el props es setCategories hook
      setInputValue(''); //limpia la caja de texto y se puede poner antes, por que el componente tiene prioridad con la propiedad 
      // port eso se ejecuta primero el onNewCategory y luego la funcion del useState hook y este proceso es mas rapido. 
      onNewCategory(inputValue.trim()); //esta ves es pasando el prop la funcion creada en el padre. 
   }

   return (
      // <form onSubmit={(event) => onSubmit(event)}>
      <form onSubmit={onSubmit}>
         <input
            type="text"
            placeholder="Buscar Gifs"
            value={inputValue}
            onChange={(event) => onInputChange(event)}
         />
      </form>
   )
}

AddCategory.propTypes = {
   onNewCategory: PropTypes.func.isRequired,
}