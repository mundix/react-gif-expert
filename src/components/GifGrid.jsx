export const GifGrid = ({category}) => {
   // Se vamos a tener codigo par alguna logia, si se usa el return e ves del implicito.
  return (
    <>
      <h3>{category}</h3>
      <p>Hola Mundo</p>
    </>
  )
}
