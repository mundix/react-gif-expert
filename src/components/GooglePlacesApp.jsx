import { useState } from 'react';
import GooglePlacesAutocomplete from 'react-google-places-autocomplete';

export const GooglePlacesApp = () => {
   const [value, setValue] = useState(null)
   const api_key="AIzaSyBx_cZFCH6pbRCdWKf8WopN5bBS6LiBIA4";
  return (
   <div>
    <GooglePlacesAutocomplete
      apiKey={api_key}
      selectProps={{
         value,
         onChange: setValue
      }}
    />
  </div>
  )
}
