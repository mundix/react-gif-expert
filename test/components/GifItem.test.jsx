import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas con <GifItem/>', () => {

   const title = 'saitama';
   const url = 'htps://one-punch/saitama.jpg';

   test('Debe hacer match con el snapshot', () => {

      const { container } = render(<GifItem title={title} url={url} />);
      expect(container).toMatchSnapshot();
   });

   test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

      render(<GifItem title={title} url={url} />);
      // screen.debug();
      // console.log(screen.getByRole('img').src);
      // expect(screen.getByRole('img').src).toBe(url);
      // expect(screen.getByRole('img').alt).toBe(title);

      const { src, alt } = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);

   })

   test('Debe de mostrar el titulo de mi componetne', () => {

      render(<GifItem title={title} url={url} />);

      // expect(screen.getByRole('p').innerHTML).toBe(title);
      // console.log(screen.getByRole('p').innerHTML);
      expect(screen.getByText( title)).toBeTruthy();

   })

})