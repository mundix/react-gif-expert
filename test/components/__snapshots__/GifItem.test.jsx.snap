// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="htps://one-punch/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
